### PR TITLE
[IN-22] Change app interface language

### PR DESCRIPTION
**Pull request by adamjorge**
Linked to this ticket: https://instamint-deva.atlassian.net/browse/IN-22

Summary of the PR:

- Handling app language using next-intl i18n lib (ICU compatible)
- Refacto of providers
- Segmentation of the admin layout and the main app layout logic
- Translation in 6 languages made using i18n nexus

### CODE REVIEW

File: src/app/[locale]/search/page.tsx [hunk 1]
```diff
@@ -16,26 +18,26 @@ export default function SearchPage() {
   if (!search) {
     return (
       <div className="text-center">
-        <p>Welcome to your search page. This is where you can search for NFTs or minters.</p>
+        <p>Search</p>
```
Comment (jimmy-martin) [inline src/app/[locale]/search/page.tsx]: maybe you forgot to translate here ?

**Review by jimmy-martin — CHANGES_REQUESTED** [review]

**Review by jimmy-martin — COMMENTED** [review]
Update tests too

File: src/app/[locale]/search/page.tsx (hunk 1 above)
Comment (adamjorge) [reply src/app/[locale]/search/page.tsx]: Yes, changed the locales json structure to translate here

File: src/app/[locale]/search/loading.tsx [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { useTranslations } from "next-intl"
+
+export default function Loading() {
+  const t = useTranslations()
+
+  return <div className="text-center">{t("loading")}</div>
```
Comment (jimmy-martin) [inline src/app/[locale]/search/loading.tsx]: you have to load "global" to reach `t("loading")`

**Review by jimmy-martin — CHANGES_REQUESTED** [review]

**Review by jimmy-martin — APPROVED** [review]

File: src/components/custom/minters/minter-list.tsx [hunk 3]
```diff
@@ -1,12 +1,15 @@
-import { MinterSearchMintersSchemaType } from "@/validators/schemas/search/minters/minterSearchMinterSchema"
 import MinterCard from "@/components/custom/minters/minter-card"
+import { MinterSearchMintersSchemaType } from "@/validators/schemas/search/minters/minterSearchMinterSchema"
+import { useTranslations } from "use-intl"
 
 export default function MinterList({ minters }: MinterListProps) {
+  const t = useTranslations("search")
+
   return (
     <div className="my-1">
```
Comment (avetisk) [inline src/components/custom/minters/minter-list.tsx:9]: Margins should be affected from outside.

Also, you should always forward `className` prop to the outermost element of your component (which will allow to apply margins for instance).

**Pull request merged**